### PR TITLE
Voidwalker now can upgrades. Also 3 new voidwalker spells.

### PR DESCRIPTION
**Pull request by Xackii**
## About The Pull Request

Voidwalker now can upgrade himself, his weapons and his skills for the points he receives when he kidnaps people.
Voidwalker starts with 1 point at the beginning and can spend it on improving the antagonist panel.
2024-11-1717-36-59-ezgif com-video-to-gif-converter
Upgrades divided into 6 categories and also into their tiers, that is, in order to gain any of the upgrades further, you first need to gain the upgrade in front of it (left of it). Also in this panel, voidwalker can open new spells (2 points for each) and upgrade them in the future. 
2024-11-1717-37-09-ezgif com-video-to-gif-converter
There is 3 new spells thats not given in roundstart and need to be unlocked (it costs 2 points):
2024-11-1717-37-40-ezgif com-video-to-gif-converter

1.  Through The Void: Cooldown 30 seconds. After 3 seconds teleports you behind the target.
2024-11-1717-38-50-ezgif com-video-to-gif-converter

2. Space Relocation: Cooldown 30 seconds. Creates a 3x3 tiles of space around voidwalker for 15 seconds.
2024-11-1717-39-59-ezgif com-video-to-gif-converter

3. Void Symbol: Cooldown 5 seconds. Can be used in space dive. Used for two purposes: Use Void Symbol on voided crewmambers to make them void blessed. You can communicate with them via transmit. Use Void Symbol on normal crewmember and all void blessed friends will gain objective to kill this crewmember.

Void Blessed is new minion antag that voidwalker can make using a void symbol on a person with a void trauma. Void Blessed peoples obey you, get a weak version of void eater and also cease to be pacifists and lose cosmophobia. They take no damage from space but still cannot fly around it. They can kidnap peoples like voidwalker but only that one that voidwalker marked with void symbol. When voidwalker mark somone with void symbol all void blessed peoples that obey to this voidwalker will gain navigator to the target and objective to kidnap him.

Here all upgrades and it's effects:
Снимок экрана 2024-11-17 180744
Bloodliness I: Increases Void Eater damage +5 and bare wound bonus +20.
Bloodiness II: Increases Void Eater damage +5 and bare wound bonus +10. Your attacks now heals you +20% of caused damage
Glass Appendage: Increases Void Eater damage +15 but also increases damage lose per hit at x10. Increases minimum damage +5
Protection: Gives you +25% basic block chance
Transfusion I: In addition to reducing damage, each hit also increases block chance +5%(Max 80%)
Transfusion II In addition to reducing damage, each hit also increases block chance +5%(Max 80%). Gives you +15% basic block chance
Accuracy I: Increases armour penetration +25% and wound bonus +30%
Silence: Attacks temporarily mute the target
Accuracy II: Increases armour penetration +25% and wound bonus +30%. Your attacks now heals you +10% of caused damage
Снимок экрана 2024-11-17 181115
Survival I: Heals +0.5 hp per tick in space (1.5 deafault). x2 heals on crit
Survival II: Heals +1 hp per tick in space (1.5 deafault). Also Heals +1hp per tick out of space
Survival III: Heals +0.5 hp per tick (1.5 deafault). x3 heals on crit. Also Heals +1hp per tick out of space
Body Robust I: Increases body's resistance to all types of damage by 25%
Strong Arms: Your arms become overgrown with muscles, which allows you to drag and throw any objects
Body Robust II: Increases body's resistance to all types of damage by 25%
Back Adaptation: Allows you to put an item which other people can put on their back on your back
Hands Adaptation: Allows your hands to use things you couldn't use when your fingers were big
Снимок экрана 2024-11-17 181407
Refraction I: Increases stun time and reduces cast time by 2 seconds
Jubilation: Speeds up by 6 seconds after using a spell
Refraction II: Increases stun time and reduces cast time by 2 seconds
Unsettle Cooldown I: Reduces cooldown time by 4 seconds
Unsettle Cooldown II: Reduces cooldown time by 4 seconds
Devastation: When using a spell, all targets within a 2 radius of the victim receive a knockdown equal to half the paralysis time and half the stamina damage
Снимок экрана 2024-11-17 181557
Through The Void Cooldown I: Reduces cast time by 5 seconds
Horror I: Slows down the target when you teleport to
Horror II: Creates 1-3 illusions near you on teleport. Number depends on the people nearby
Impatience: Removes cast delay
Through The Void Cooldown II: Reduces cast time by 10 seconds
Снимок экрана 2024-11-17 181745
Space Relocation Cooldown I: Increases spell effect duration by 5 seconds and cooldown by 10
Economy: You can use spell again to deactivate it early and half cooldown timer
Space Relocation Cooldown II: Increases spell effect duration by 10 seconds and cooldown by 20
Dissolution: Makes all windows, walls and doors in range translucent allowing everyone to pass through them
Expansion: Icreases space creating around to 5x5
Снимок экрана 2024-11-17 181933
Influence I: +1 Max void blessed person that you can make
Influence II: +1 Max void blessed person that you can make
Influence III: +2 Max void blessed person that you can make
Inheritance: All void blessed followers get unsettle spell
Promotion: Targets kidnapped by void blessed followers will give 2 points instead of 1

Due fact that there are too many upgrades (and you most likely won’t be able to study them all), up to 6 random upgrades can become free (they are indicated by a brain icon in front of the upgrade name). This should make upgrade choosing direction a little easier.

Other additions:
Voidwalker now have roundend report where loging kidnapped targets counter, all researched spells and upgrades and information about void blesseds and his objectives.
Fixed glass pacifist, no crit hitting and only pull living elements don't removing when you remove voidwalker datum from mob.
New sprite for unsettle.
Fixed that succesfully_kidnapped proc don't takes info about kidnapped victim.

## Why It's Good For The Game
Despite the fact that the voidwalker is a midround antagonist, it is still an antagonist and has several limitations that do not allow you to come up with a variety of gameplay and there is no development. PR adds the ability to upgrade your voidwalker by choosing directions between getting new spells, doing void blesseds, or improve your body to get rid of some limitations, such as the fact that you cannot pull items, wear a backpack, or shoot laser.

## Changelog

:cl:
add: Voidwalker now can upgrades.
add: 3 new spells for Voidwalker.
/:cl:

### CODE REVIEW

**Comment by fineks** [comment]
void antag, heretic path

**Comment by carlarctg** [comment]
this is so cool that i really wish it was good. try to make your own antag?

**Comment by Jacquerel** [comment]
I'll be honest I don't think it's desirable for every antagonist to have a progression path with increased power, some things just being what they are immediately is good.

**Comment by Rex9001** [comment]
I think this is pretty cool, and has some very nice ideas but I echo jacqs thoughts, maybe you could limit the upgrade stuff and just give a set amount of points at the start? So it isnt a progression antag but has opportunity to start off different based on the players preferred playstyle

**Comment by Time-Green** [comment]
There is no variant of this that will get merged.
I am okay with new abilities, something akin to rerollable blob strains, but this goes so far beyond what the voidwalker is, it's better to just make a new antag instead.

Voidwalker is not a main-stage antagonist, unlike wizards, blobs, space dragons. I also don't wish to let them become main stage antagonists like heretics or some traitors and changelings can do.

Biddle is also not something I wish to see in Voidwalker. Straight up buffing damage, armor, spells etc. The brainwash especially is so far outside of the Voidwalkers design space that it looks more like a completely different antagonist. This is a solitary hunter, not a social conversion antag!

I really wish you had discussed this with us before hand, I could've told you immediately not to make this PR.\

The UI looks cool though, and there's some creative abilities, but I don't wish to see voidwalker become a heretic-like snowballing machine. Maybe roll these into a new antagonist?

**Comment by Time-Green** [comment]
God I really wish you talked to us

**Comment by MoonDeer728** [comment]
> There is no variant of this that will get merged. I am okay with new abilities, something akin to rerollable blob strains, but this goes so far beyond what the voidwalker is, it's better to just make a new antag instead.
> 
> Voidwalker is not a main-stage antagonist, unlike wizards, blobs, space dragons. I also don't wish to let them become main stage antagonists like heretics or some traitors and changelings can do.
> 
> Biddle is also not something I wish to see in Voidwalker. Straight up buffing damage, armor, spells etc. The brainwash especially is so far outside of the Voidwalkers design space that it looks more like a completely different antagonist. This is a solitary hunter, not a social conversion antag!
> 
> I really wish you had discussed this with us before hand, I could've told you immediately not to make this PR.\
> 
> The UI looks cool though, and there's some creative abilities, but I don't wish to see voidwalker become a heretic-like snowballing machine. Maybe roll these into a new antagonist?

no

**Comment by Time-Green** [comment]
I can't stress this enough, but please talk with us before undertaking such massive changes. Please please please talk with us beforehand

**Comment by Xackii** [comment]
> I can't stress this enough, but please talk with us before undertaking such massive changes. Please please please talk with us beforehand

then there won't be a surprise

**Comment by Xackii** [comment]
> There is no variant of this that will get merged. I am okay with new abilities, something akin to rerollable blob strains, but this goes so far beyond what the voidwalker is, it's better to just make a new antag instead.
> 
> Voidwalker is not a main-stage antagonist, unlike wizards, blobs, space dragons. I also don't wish to let them become main stage antagonists like heretics or some traitors and changelings can do.
> 
> Biddle is also not something I wish to see in Voidwalker. Straight up buffing damage, armor, spells etc. The brainwash especially is so far outside of the Voidwalkers design space that it looks more like a completely different antagonist. This is a solitary hunter, not a social conversion antag!
> 
> I really wish you had discussed this with us before hand, I could've told you immediately not to make this PR.\
> 
> The UI looks cool though, and there's some creative abilities, but I don't wish to see voidwalker become a heretic-like snowballing machine. Maybe roll these into a new antagonist?

Even though I see that you are strongly against my PR, I’ll still put in my 5 cents. Despite the fact that it looks very much like a heretic, it is a little different. Upgrades for a voidwalker do not give serious changes, do not end the round and do not force the voidwalker to worry about every point. Voidwalker can calmly choose improvements for himself depending on the situation or his own preferences. I could compare it to revenant but a little deeper. It's just that at the moment there are few ways for a Voidwalker to have fun. you have only 1 spell that takes a long time to charge and gives a small stun, and you need to fly through space and try to catch at least 1 target. and when do you do this? you don't get anything for it. it feels more like you are griefing the round by making the players silent and pacifiс and you still don’t get anything in return and can’t do anything interesting due to restrictions. one way or another, if you are not interested in this and you think it is better to add a new antag, then I will do so.

**Comment by moocowswag** [comment]
What if another antag could spend a bit of their own resource pool to make void walker a stronger threat, like a wizard or void heretic.

**Comment by Xackii** [comment]
> What if another antag could spend a bit of their own resource pool to make a void walker a stronger threat, like a wizard or void heretic.

Voidwalker rolling very rarely (at least I saw him only 1 time).

**Comment by moocowswag** [comment]
> > What if another antag could spend a bit of their own resource pool to make a void walker a stronger threat, like a wizard or void heretic.
> 
> Voidwalker rolling very rarely (at least I saw him only 1 time).

Even more likely to get merged then because its an exceedingly rare scenario

**Comment by Time-Green** [comment]
> > I can't stress this enough, but please talk with us before undertaking such massive changes. Please please please talk with us beforehand
> 
> then there won't be a surprise

I respect your positive outlook. It's also okay to just DM a maintainer if you want to keep the surprise up, although I generally recommend you just open up a discord discussion thread. The real surprise is if someone ends up doing!

**Comment by Loafin34** [comment]
biddle nightmare

**Review by Wallemations — APPROVED** [review]
I know there's a DNM, but just in case that gets lifted or whatever, this is **_just_** an approval of the sprites that were added, which I feel are all very good. 
I can't say anything about the balance or whatever.

File: tgui/packages/tgui/interfaces/AntagInfoVoidwalker.tsx [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+import {
+  BlockQuote,
+  Button,
+  DmIcon,
+  LabeledList,
+  Section,
+  Stack,
+  Table,
+  Tabs,
+} from '../components';
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/AntagInfoVoidwalker.tsx:13]:
```suggestion
} from 'tgui-core/components';
```

File: tgui/packages/tgui/interfaces/AntagInfoVoidwalker.tsx [hunk 2]
```diff
@@ -9,63 +21,241 @@ const noticestyle = {
...
-                <LabeledList.Item label="Unsettle">
-                  Target a victim while remaining only partially in their view
-                  to stun and weaken them, but also announce them your presence.
-                </LabeledList.Item>
-              </LabeledList>
-            </Section>
+          <Stack.Item grow>
+            {(currentTab === 0 && <VoidwalkerIntroduction />) || (
+              <VoidwalkerUpgrades />
+            )}
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/AntagInfoVoidwalker.tsx:258]: ternary please

File: tgui/packages/tgui/interfaces/AntagInfoVoidwalker.tsx [hunk 3]
```diff
@@ -9,63 +21,241 @@ const noticestyle = {
...
+                    my={6}
+                    mx={4}
+                    color="purple"
+                    onClick={() => {
+                      act('unlock', {
+                        what_unlock: upgrade_box.name,
+                      });
+                    }}
+                  />
+                )}
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/AntagInfoVoidwalker.tsx:220]: this is a pretty big conditional. can you make this into two components please?

**Pull request closed without merging**